Dockerfile

```diff
@@ -1,9 +1,22 @@
-FROM python:latest
+FROM python:3 AS compile-image
 MAINTAINER Julian-Samuel Gebühr
 
-RUN apt-get update && apt-get install -y pip
-RUN /usr/local/bin/python -m pip install --upgrade pip
-RUN pip install matrix-registration-bot
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends build-essential gcc
+
+RUN python -m venv /opt/venv
+# Make sure we use the virtualenv:
+ENV PATH="/opt/venv/bin:$PATH"
+
+RUN pip install --no-cache-dir matrix-registration-bot
+
+FROM python:3-slim
+
+COPY --from=compile-image /opt/venv /opt/venv
```

Comment:
Copying over virtualenvs might be problematic if the locally stored images `python:3` and `python:3-slim` have different Python versions. Due to caching or bad timing this issue might actually occur and would result in a broken image. Any idea how to handle this case?